dbctx/soup.go

```diff
@@ -0,0 +1,83 @@
+package dbctx
+
+import (
+	"time"
+
+	"github.com/jinzhu/gorm"
+)
+
+type (
+	soupRepository struct {
+		connFactory ConnectionFactory
+	}
+	// SoupRepository to provide access to methods to handle soup data
+	SoupRepository interface {
+		// SetSoup sets the soup of the day
+		SetSoup(name string) error
+		GetSoupOfTheDay() (*SoupOfTheDay, error)
+	}
+)
+
+// NewSoupRepository creates a new repository
+func NewSoupRepository(connFactory ConnectionFactory) SoupRepository {
+	return &soupRepository{
+		connFactory,
+	}
+}
+
+func (m *soupRepository) SetSoup(name string) error {
```

Comment:
Consider renaming this to `SetSoupOfTheDay`

Reply:
Yeah - good point